official/projects/lra/transformer_encoder.py

```diff
@@ -0,0 +1,288 @@
+# Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Transformer encoder."""
+
+# pylint: disable=g-classes-have-attributes
+
+from typing import Any, Callable, List, Optional, Union
+
+from absl import logging
+import tensorflow as tf
+
+from official.modeling import tf_utils
+from official.nlp.modeling import layers
+
+_Initializer = Union[str, tf.keras.initializers.Initializer]
+_approx_gelu = lambda x: tf.keras.activations.gelu(x, approximate=True)
+
+
+class TransformerEncoder(tf.keras.layers.Layer):
+  """TransformerEncoder.
```

Comment:
I feel we need docstring to explain what's the difference of this guy v.s. the bert?

Reply:
They are mostly similar, with small adaptations for the pipeline. If you think we need a docstring I'll update that.